src/overlays/actors/ovl_Bg_Keikoku_Spr/z_bg_keikoku_spr.c

```diff
@@ -20,14 +25,38 @@ const ActorInit Bg_Keikoku_Spr_InitVars = {
     (ActorFunc)BgKeikokuSpr_Init,
     (ActorFunc)BgKeikokuSpr_Destroy,
     (ActorFunc)BgKeikokuSpr_Update,
-    (ActorFunc)BgKeikokuSpr_Draw
+    (ActorFunc)BgKeikokuSpr_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Keikoku_Spr/BgKeikokuSpr_Init.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 3000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 200, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 400, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 20, ICHAIN_STOP),
+};
+
+void BgKeikokuSpr_Init(Actor* thisx, PlayState* play) {
+    BgKeikokuSpr* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+}
+
+void BgKeikokuSpr_Destroy(Actor* thisx, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Keikoku_Spr/BgKeikokuSpr_Destroy.s")
+void BgKeikokuSpr_Update(Actor* thisx, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Keikoku_Spr/BgKeikokuSpr_Update.s")
+void BgKeikokuSpr_Draw(Actor* thisx, PlayState* play) {
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Keikoku_Spr/BgKeikokuSpr_Draw.s")
+    OPEN_DISPS(play->state.gfxCtx);
+    AnimatedMat_Draw(play, Lib_SegmentedToVirtual(object_keikoku_obj_Matanimheader_0001F8));
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_XLU_DISP++, object_keikoku_obj_DL_000100);
+    AnimatedMat_Draw(play, Lib_SegmentedToVirtual(object_keikoku_obj_Matanimheader_0003F8));
+    gSPDisplayList(POLY_XLU_DISP++, object_keikoku_obj_DL_000300);
+    AnimatedMat_Draw(play, Lib_SegmentedToVirtual(object_keikoku_obj_Matanimheader_0005F8));
+    gSPDisplayList(POLY_XLU_DISP++, object_keikoku_obj_DL_000500);
+    CLOSE_DISPS(play->state.gfxCtx);
```

Comment:
```suggestion
    OPEN_DISPS(play->state.gfxCtx);
    
    AnimatedMat_Draw(play, Lib_SegmentedToVirtual(object_keikoku_obj_Matanimheader_0001F8));
    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
    gSPDisplayList(POLY_XLU_DISP++, object_keikoku_obj_DL_000100);
    AnimatedMat_Draw(play, Lib_SegmentedToVirtual(object_keikoku_obj_Matanimheader_0003F8));
    gSPDisplayList(POLY_XLU_DISP++, object_keikoku_obj_DL_000300);
    AnimatedMat_Draw(play, Lib_SegmentedToVirtual(object_keikoku_obj_Matanimheader_0005F8));
    gSPDisplayList(POLY_XLU_DISP++, object_keikoku_obj_DL_000500);
    
    CLOSE_DISPS(play->state.gfxCtx);
```
newlines separating the disps macros are preferred.